apps/factory/src/components/marketplace/like_button.tsx

```diff
@@ -1,14 +1,76 @@
-import React, { useState } from "react";
+import React, { useEffect, useState } from "react";
 import Button from "@mui/material/Button";
 import ButtonGroup from "@mui/material/ButtonGroup";
 import FavoriteIcon from "@mui/icons-material/Favorite";
 
-function LikeButton({ count }: { count: number }) {
-  const [counter, setCounter] = useState(count);
+import { api } from "~/utils/api";
+import { signIn, useSession } from "next-auth/react";
 
-  const handleLikeClick = () => {
-    // Increment the like count when the button is clicked
-    setCounter(counter + 1);
+interface LikeButtonProps {
+  EntityId: string;
+  EntityType: string;
+}
+
+const LikeButton: React.FC<LikeButtonProps> = ({ EntityId, EntityType }) => {
+  const { data } = api.like.getLikes.useQuery({
+    EntityId,
+    EntityType,
+  });
+
+  const [counter, setCounter] = useState<number>(
+    data?.likesCount != null ? data.likesCount : 0,
+  );
+  const [hasLiked, setHasLiked] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const [LikeId, setLikeId] = useState<string>("");
+
+  useEffect(() => {
+    data?.likes.some((like) => {
+      if (like.userId === sessionData?.user.id) {
+        setHasLiked(true);
+        setLikeId(like.likeId!);
+      }
+    });
+
+    if (data?.likesCount != null) {
+      setCounter(data.likesCount);
+    }
+  }, [data?.likesCount]);
+
+  const UpdateMutation = api.like.createLike.useMutation();
+
+  const { data: sessionData } = useSession();
+  const handleLikeClick = async () => {
+    setLoading(true);
+    console.log(data?.likesCount);
+    console.log(data?.likes);
+
+    UpdateMutation.mutate(
+      {
+        EntityId,
+        EntityType,
+        hasLiked,
+        LikeId,
+      },
+      {
+        onSuccess() {
+          console.log("Likes Updated");
+        },
+        onError(error) {
+          const errorData = JSON.parse(error.message);
+          console.log("error: ", errorData);
+        },
+      },
+    );
+
+    if (hasLiked) {
+      setCounter(counter! - 1);
+      setHasLiked(!hasLiked);
+    } else {
+      setCounter(counter! + 1);
+      setHasLiked(!hasLiked);
```

Comment:
We need to discuss this. The hasLiked should not be sent back to the like. Like action should be done and completlely handled with out hasLiked information from frontend.